addons/event_sale/static/src/js/event_configurator_widget.js

```diff
@@ -0,0 +1,162 @@
+            this.$('.o_event_sale_js_event_configurator_edit').length === 0) {
+            var $editConfigurationButton = $('<button>', {
+                type: 'button',
+                class: 'fa fa-pencil btn btn-secondary o_event_sale_js_event_configurator_edit',
+                tabindex: '-1',
+                draggable: false,
+                'aria-label': _t('Edit Configuration'),
+                title: _t('Edit Configuration')
+            });
+
+            $inputDropdown.after($editConfigurationButton);
+        }
+
+        return prom;
+    },
+
+    /**
+     * This method is overridden to check if the product_id needs configuration or not:
+     *
+     * @override
+     * @param {OdooEvent} event
+     *
+     * @private
+     */
+    _onFieldChanged: function (event) {
+        var self = this;
+
+        this._super.apply(this, arguments);
+        if (!event.data.changes.product_id){
+            return;
+        }
+
+        var product_id = event.data.changes.product_id.id;
+        if (product_id){
+            this._rpc({
+                model: 'product.product',
+                method: 'is_event',
+                args: [
+                    product_id
+                ]
+            }).then(function (isEvent){
+                if (isEvent){
+                    self._openEventConfigurator({
+                            default_product_id: product_id
+                        },
+                        event.data.dataPointID
+                    );
+                }
+            });
+        }
+    },
+
+    /**
+     * Opens the event configurator in 'edit' mode.
+     *
+     * @private
+     */
+    _onEditEventConfiguration: function () {
+        var defaultValues = {
+            default_product_id: this.recordData.product_id.data.id
+        };
+
+        if (this.recordData.event_id) {
+            defaultValues['default_event_id'] = this.recordData.event_id.data.id;
+        }
+
+        if (this.recordData.event_ticket_id) {
+            defaultValues['default_event_ticket_id'] = this.recordData.event_ticket_id.data.id;
+        }
+
+        this._openEventConfigurator(defaultValues, this.dataPointID);
+    },
+
+    /**
+     * Opens the event configurator to allow configuring the SO line with events information.
+     *
+     * When the window is closed, configured values are used to trigger a 'field_changed'
+     * event to modify the current SO line.
+     *
+     * If the window is closed without providing the required values 'event_id' and
+     * 'event_ticket_id', the product_id field is cleaned.
+     *
+     * @param {Object} data various "default_" values
+     * @param {string} dataPointId
+     *
+     * @private
+     */
+    _openEventConfigurator: function (data, dataPointId) {
+        var self = this;
+        self._rpc({
+            model: 'ir.model.data',
+            method: 'xmlid_to_res_id',
+            kwargs: {xmlid: 'event_sale.event_configurator_view_form'},
```

Comment:
Not sure how I can do that, would love an example if you can provide me one?

Reply:
https://github.com/odoo/odoo/blob/master/addons/mail/static/src/js/systray/systray_messaging_menu.js#L143 even if here, `mail_discuss` is a client action, it will work as client action is an action.

Reply:
Saves code, rpc, and makes tests clearer. Thanks!